core/src/main/java/org/pentaho/di/core/Const.java

```diff
@@ -0,0 +1,4059 @@
+    int len = searchFor.length();
+    if ( len > 0 ) {
+      int start = string.indexOf( searchFor );
+      while ( start != -1 ) {
+        counter++;
+        start = string.indexOf( searchFor, start + len );
+      }
+    }
+    return counter;
+  }
+
+  public static String[] GetAvailableFontNames() {
+    GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+    Font[] fonts = ge.getAllFonts();
+    String[] FontName = new String[fonts.length];
+    for ( int i = 0; i < fonts.length; i++ ) {
+      FontName[i] = fonts[i].getFontName();
+    }
+    return FontName;
+  }
+
+  public static String getKettlePropertiesFileHeader() {
+    StringBuilder out = new StringBuilder();
+
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line01", BuildVersion
+      .getInstance().getVersion() )
+      + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line02" ) + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line03" ) + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line04" ) + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line05" ) + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line06" ) + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line07" ) + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line08" ) + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line09" ) + CR );
+    out.append( BaseMessages.getString( PKG, "Props.Kettle.Properties.Sample.Line10" ) + CR );
+
+    return out.toString();
+  }
+
+  /**
+   * Mask XML content. i.e. protect with CDATA;
+   *
+   * @param content
+   *          content
+   * @return protected content
+   */
+  public static String protectXMLCDATA( String content ) {
+    if ( Utils.isEmpty( content ) ) {
+      return content;
+    }
+    return "<![CDATA[" + content + "]]>";
+  }
+
+  /**
+   * Get the number of occurrences of searchFor in string.
+   *
+   * @param string
+   *          String to be searched
+   * @param searchFor
+   *          to be counted string
+   * @return number of occurrences
+   */
+  public static int getOcuranceString( String string, String searchFor ) {
+    if ( string == null || string.length() == 0 ) {
+      return 0;
+    }
+    Pattern p = Pattern.compile( searchFor );
+    Matcher m = p.matcher( string );
+    int count = 0;
+    while ( m.find() ) {
+      ++count;
+    }
+    return count;
+  }
+
+  /**
+   * Mask XML content. i.e. replace characters with &values;
+   *
+   * @param content
+   *          content
+   * @return masked content
+   */
+  public static String escapeXml( String content ) {
+    if ( Utils.isEmpty( content ) ) {
+      return content;
+    }
+    return StringEscapeUtils.escapeXml( content );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

escapeJava(), escapeHtml3(), escapeHtml4(), escapeXml(), escapeXml10(), and escapeXml11() have been deprecated in org.apache.commons.lang.StringEscapeUtils and org.apache.commons.lang3.StringEscapeUtils and do not cover the entire range of characters that have to be encoded. OWASP Java Encoder library should be used instead.